File: Dante2/Module2.07/Zadanie1.02/main.c
#include <stdio.h>
#include "int_operations.h"

int main()
{
    int error;

    int a, b;
    printf("Enter: ");
    error = scanf("%i %i",&a, &b);
    if(error != 2)
    {
        printf("Incorrect input");
        return 1;
    }

    int type;
    printf("Enter type: ");
    error = scanf("%i",&type);
    if(error != 1)
    {
        printf("Incorrect input");
        return 1;
    }
    if(type < 0 || type > 3)
    {
        printf("Incorrect input data");
        return 2;
    }

    add_int(a, b);
    sub_int(a, b);
    mul_int(a, b);
    div_int(a, b);

    printf("%i", get_function(type)(a, b));

    return 0;
}

File: Dante2/Module2.07/Zadanie1.02/main.rs
use crate::int_operations::{get_function};
mod int_operations;

fn main() {
    let mut input: String = String::new();

    input.clear();
    println!("Enter a: ");
    std::io::stdin().read_line(&mut input).expect("Incorrect input");
    let a: i32 = input.trim().parse().expect("Incorrect parse");

    input.clear();
    println!("Enter b: ");
    std::io::stdin().read_line(&mut input).expect("Incorrect input");
    let b: i32 = input.trim().parse().expect("Incorrect parse");

    input.clear();
    println!("Enter type: ");
    std::io::stdin().read_line(&mut input).expect("Incorrect input");
    let t: i32 = input.trim().parse().expect("Incorrect parse");

    println!("Result is: {}", get_function(t)(a, b));
}
